### PR TITLE
add "reset to default" color option for ColorPicker

### LINKED ISSUE #8642: [ColorPicker] Add the option to reset to the default color
### Is your proposal related to a problem?

Currently, there is no option to reset the color to the default value while selecting the "Brand Color". While experimenting with the colors, the user might want to reset back to the default color.

https://user-images.githubusercontent.com/47350983/235972896-4f1f37d0-03bc-4228-8581-a3f2fe02e8e8.mp4

### Describe the solution you'd like

 It would be user-friendly to have a reset option next to the input box, which will set the color back to the default value.

### PR DESCRIPTION
**Pull request by Praneeth2k**
## What does this PR do?

Fixes #8642
Adds the option to reset to the default color.

https://user-images.githubusercontent.com/47350983/235989553-07d3de0b-e109-4f40-bd9e-0d9d1132893f.mp4

Description of the code changes:

ColorPicker
- Add a non-required prop to the props passed on the the component called "resetDefaultValue".
- Add a button which has the RotateCcw icon. This icon is invisible by default. Only if the resetDefaultValue is passed on by the part component, this icon will be visible.
- Upon clicking this icon, the color is set to the "resetDefaultValue"

The color picker component is used in 3 places: 
1. Appearance section for a user: https://app.cal.com/settings/my-account/appearance - 2 instances.
   Default resetDefaultValue of #292929 and #fafafa added for light and dark themes .
2. Appearance section for a Team - 2 instances.
   Default resetDefaultValue of #292929 and #fafafa added for light and dark themes .
3. The Embed option for a meeting - 1 instance.
  resetDefaultValue has not been set meaning the reset icon will not appear.

**Environment**: Staging(main branch) / Production

## Type of change

- New feature (non-breaking change which adds functionality)

## How should this be tested?

- [x] Set the color to a different value and then use the reset option.
- [x] Click on Update at the bottom without using the reset option and refresh page to ensure that the color is saved.
- [x] Click on Update at the bottom after using the reset option and refresh page to ensure that the color is saved.

### CODE REVIEW

**Review by Udit-takkar — COMMENTED** [review]
@Praneeth2k some type checks are failing. can you fix them ?

File: packages/ui/components/form/color-picker/colorpicker.tsx [hunk 1]
```diff
@@ -57,6 +62,21 @@ const ColorPicker = (props: ColorPickerProps) => {
...
+          variant="icon"
+          rel="noreferrer"
+          StartIcon={RotateCcw}
+          tooltip="Reset to default"
+          onClick={() => {
+            setColor(fallBackHex(resetDefaultValue, false));
+            props.onChange(resetDefaultValue);
+          }}
+        />
+      </div>
```
Comment (sean-brydon) [inline packages/ui/components/form/color-picker/colorpicker.tsx:79]:
```suggestion
     {!resetDefaultValue &&  <div className={classNames("px-1"}>
        <Button
          color={resetDefaultValue == "#292929" ? "primary" : "secondary"}
          data-testid="preview-button"
          target="_blank"
          variant="icon"
          rel="noreferrer"
          StartIcon={RotateCcw}
          tooltip="Reset to default"
          onClick={() => {
            setColor(fallBackHex(resetDefaultValue, false));
            props.onChange(resetDefaultValue);
          }}
        />
      </div>} 
```

We ideally want this not to show in the dom.

Comment (Praneeth2k) [reply packages/ui/components/form/color-picker/colorpicker.tsx]: I've incorporated this change with one modification. 
It should be `{resetDefaultValue && ...` and not `{!resetDefaultValue && ...`

@Udit-takkar This has fixed the type-checks.

**Comment by sean-brydon** [comment]
Thanks for fixing the issues addressed - The reset button doesnt seem to hide when the values are default

**Comment by Praneeth2k** [comment]
> The reset button doesnt seem to hide when the values are default

@sean-brydon, Just to clarify, what I was trying to achieve is that if the "resetDefaultValue" was not passed on to the component then there is no value to reset to, hence the button should not show up. 

I agree that it would be nice if we do not show the reset button if the color is already set to the default value. I have pushed the changes to implement this. 

Thanks for the suggestion!

**Review by sean-brydon — APPROVED** [review]
Love it :) Thanks bud!

File: packages/ui/components/form/color-picker/colorpicker.tsx [hunk 2]
```diff
@@ -57,6 +62,21 @@ const ColorPicker = (props: ColorPickerProps) => {
         }}
         type="text"
       />
+      <div className={classNames("px-1", resetDefaultValue == null && "invisible")}>
+        <Button
+          color={resetDefaultValue == "#292929" ? "primary" : "secondary"}
+          data-testid="preview-button"
```
[the remark this answers is not in the document]
Comment (Udit-takkar) [reply packages/ui/components/form/color-picker/colorpicker.tsx]: @Praneeth2k yeah remove this

**Pull request merged**